### ISSUE TITLE
[dev.icinga.com #11226] Host/Service not scheduled according to retry_interval when in hard soft state

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#3972]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11226

**Created by barry.quiel on 2016-02-25 00:47:38 +00:00**

Assignee: _(none)_
Status: _Closed_  (closed on _2016-08-08 15:34:49 +00:00_)
Target Version: _(none)_
Last Update: _2016-11-09 14:59:54 +00:00 (in Redmine)_

    Icinga Version: 2.4.3
    Backport?: Not yet backported
    Include in Changelog: 1

---

I was hoping this issue would be fixed in [\#7287](https://dev.icinga.com/issues/7287) but it appears to still be in this version as well.

It looks as though hosts and services are not honoring the retry\_interval for check scheduling when in a soft state. This seem to happen randomly for for hosts and services. In this case I am not using anything external ( command pipe,api ) to force the scheduling of the check.

\[1456358926\] CURRENT HOST STATE: nrt1-pop-cenet-wgw2;UP;HARD;1;PING OK - Packet loss = 0%, RTA = 96.56 ms
\[1456359599\] HOST ALERT: nrt1-pop-cenet-wgw2;DOWN;SOFT;1;PING CRITICAL - Packet loss = 100%
\[1456359616\] HOST ALERT: nrt1-pop-cenet-wgw2;DOWN;SOFT;2;PING CRITICAL - Packet loss = 100%
\[1456359748\] HOST ALERT: nrt1-pop-cenet-wgw2;DOWN;SOFT;3;PING CRITICAL - Packet loss = 100%
\[1456359755\] HOST ALERT: nrt1-pop-cenet-wgw2;DOWN;SOFT;4;PING CRITICAL - Packet loss = 100%
\[1456359830\] HOST ALERT: nrt1-pop-cenet-wgw2;DOWN;HARD;1;PING CRITICAL - Packet loss = 100%

    Object 'nrt1-pop-cenet-wgw2' of type 'Host':
      % declared in '/var/lib/icinga2/api/zones/dc-master/_etc/generated/nrt1-pop-cenet-wgw2.conf', lines 1:0-1:32
      * __name = "nrt1-pop-cenet-wgw2"
      * action_url = "'https://our.intern.facebook.com/intern/ods/chart/?submitted=1&queries[0]={"active":"true","entity":"nrt1-pop-cenet-wgw2.corp","key":"Icinga.Latency,Icinga.PacketLossPercent","source":"ods"}'"
        % = modified in '/var/lib/icinga2/api/zones/global-config/_etc/fb/odsgraph-template.conf', lines 24:4-24:62
      * address = "172.18.0.23"
        % = modified in '/var/lib/icinga2/api/zones/dc-master/_etc/generated/nrt1-pop-cenet-wgw2.conf', lines 5:4-5:26
      * address6 = "2620:10d:c060:0:172:18:0:23"
        % = modified in '/var/lib/icinga2/api/zones/dc-master/_etc/generated/nrt1-pop-cenet-wgw2.conf', lines 6:4-6:43
      * check_command = "hostalive"
        % = modified in '/var/lib/icinga2/api/zones/global-config/_etc/templates.conf', lines 19:3-19:29
      * check_interval = 240
    .....
     * package = "_cluster"
      * retry_interval = 240
        % = modified in '/var/lib/icinga2/api/zones/global-config/_etc/templates.conf', lines 17:3-17:22
        % = modified in '/var/lib/icinga2/api/zones/global-config/_etc/fb/host-template.conf', lines 6:4-6:22

You can see the check\_interval and retry\_interval as both 4 minutes (240 seconds). From the log you can see that once the host reaches the first soft state it starts checking at random intervals:
1-&gt;2 17 seconds
2-&gt;3 32 seconds
3-&gt;4 7 seconds
4-&gt;hard 75 seconds

Interestingly when it gets to the hard down state and should go back to the normal check interval of 240 seconds, the host recovers (meaning it ran another check) in 36 seconds.

**Attachments**

* [export.csv](https://dev.icinga.com/attachments/download/3153/export.csv) barry.quiel - _2016-02-29 23:31:38 +00:00_
* [debug1.log](https://dev.icinga.com/attachments/download/3378/debug1.log) barry.quiel - _2016-05-12 21:00:00 +00:00_

---

**Relations**:

* relates [#10225](https://dev.icinga.com/issues/10225)
* relates [#12180](https://dev.icinga.com/issues/12180)
